ocaml/testsuite/tests/typing-layouts/erasable_annot.ml

```diff
@@ -55,35 +64,68 @@ end;;
 Line 2, characters 2-43:
 2 |   type _ g = | MkG : ('a : immediate). 'a g
       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
-Error: Usage of layout immediate/immediate64 in g can't be erased for compatibility with upstream OCaml.
-        This error is produced due to the use of -extension-universe (no_extensions|upstream_compatible).
+Warning 187 [incompatible-with-upstream]: Usage of layout immediate/immediate64 in g can't be erased
+for compatibility with upstream OCaml. This error is produced due to
+the use of -extension-universe upstream_compatible.
+
+module type S = sig type _ g = MkG : ('a : immediate). 'a g end
 |}];;
 
 let f (type a : immediate): a -> a = fun x -> x
 [%%expect {|
 Line 1, characters 4-5:
 1 | let f (type a : immediate): a -> a = fun x -> x
         ^
-Error: Usage of layout immediate/immediate64 in f can't be erased for compatibility with upstream OCaml.
-        This error is produced due to the use of -extension-universe (no_extensions|upstream_compatible).
+Warning 187 [incompatible-with-upstream]: Usage of layout immediate/immediate64 in f can't be erased
```

Comment:
Hmmm.... two warnings. Should probably be just one.

Reply:
Fixed this particular issue, but some warnings are still duplicated.

Reply:
Let's fix this in a separate PR as discussed